src/Simulation/Simulators/CommonNativeSimulator/Dump.cs

```diff
@@ -96,5 +92,393 @@ public QSimDumpRegister(CommonNativeSimulator m) : base(m)
+        {
+            /// <summary>
+            ///     Label computational states directly by their bit strings.
+            /// </summary>
+            /// <example>
+            ///     Following this convention, the state |0⟩ ⊗ |1⟩ ⊗ |1⟩ is labeled
+            ///     by |011⟩.
+            /// </example>
+            Bitstring,
+
+            /// <summary>
+            ///     Label computational states directly by interpreting their bit
+            ///     strings as little-endian encoded integers.
+            /// </summary>
+            /// <example>
+            ///     Following this convention, the state |0⟩ ⊗ |1⟩ ⊗ |1⟩ is labeled
+            ///     by |6⟩.
+            /// </example>
+            LittleEndian,
+
+            /// <summary>
+            ///     Label computational states directly by interpreting their bit
+            ///     strings as big-endian encoded integers.
+            /// </summary>
+            /// <example>
+            ///     Following this convention, the state |0⟩ ⊗ |1⟩ ⊗ |1⟩ is labeled
+            ///     by |3⟩.
+            /// </example>
+            BigEndian
+        }
+        
+        /// <summary>
+        ///     Represents a quantum state vector and all metadata needed to display
+        ///     that state vector.
+        /// </summary>
+        public class DisplayableState
+        {
+            private static readonly IComparer<string> ToIntComparer =
+                Comparer<string>.Create((label1, label2) =>
+                    Comparer<int>.Default.Compare(
+                        Int32.Parse(label1), Int32.Parse(label2)
+                    )
+                );
+
+            /// <summary>
+            ///     Metadata to be used when serializing to JSON, allowing code
+            ///     in other languages to determine what representation is used
+            ///     for this state.
+            /// </summary>
+            [JsonProperty("diagnostic_kind")]
+            private string DiagnosticKind => "state-vector";
+
+            /// <summary>
+            ///     The indexes of each qubit on which this state is defined, or
+            ///     <c>null</c> if these indexes are not known.
+            /// </summary>
+            [JsonProperty("qubit_ids")]
+            public IEnumerable<int>? QubitIds { get; set; }
+
+            /// <summary>
+            ///     The number of qubits on which this state is defined.
+            /// </summary>
+            [JsonProperty("n_qubits")]
+            public int NQubits { get; set; }
+
+            /// <remarks>
+            ///     These amplitudes represent the computational basis states
+            ///     labeled in little-endian order, as per the behavior of
+            ///     <see cref="Microsoft.Quantum.Simulation.Simulators.CommonNativeSimulator.StateDumper.Dump" />.
+            /// </remarks>
+            [JsonProperty("amplitudes")]
+            public Complex[]? Amplitudes { get; set; }
```

Comment:
No suggested action at the moment, but when adding the sparse simulator, this will need to be generalized to a dictionary from indices to amplitudes.

Reply:
Changed to a dictionary.